lib/portlayer/network/context.go

```diff
@@ -1045,6 +1047,13 @@ func (c *Context) AddContainer(h *exec.Handle, options *AddContainerOptions) err
 				return err
 			}
 		}
+		// Check that ports are only opened on published network firewall configuration.
+		if len(options.Ports) > 0 && s.TrustLevel() != executor.Published {
```

Comment:
So if the firewall is "open" and I specify ports that is going to fail? That seems counter intuitive.